File: time/py/add-times.py
#!/usr/bin/python3

import sys

def main(argv):

    n = int (argv[0], 16)
    times = int (argv[1])
    sum = 0
    for i in range(times):
        sum = sum + n
        
    print (hex (sum))


if __name__ == '__main__':
    if len(sys.argv) != 3:
        print("usage: add NUM NUM")
        sys.exit(0)

    else:
        main(sys.argv[1:])

File: time/add-times.cpp
#include <iostream>
#include <nat.hpp>
#include <string>

int main(int argc, char ** argv)
{
  
    if (argc != 3) {
        std::cout << "Usage: add-times NUMBER TIMES" << std::endl;
        return 0;
    }

    lint::nat n{std::string{argv[1]}};
    int times{std::stoi(argv[2])};
    
    lint::nat sum;
    for (int i = 0 ; i < times; i++) {
        sum += n;
    }

    std::cout << sum << std::endl;
}
